### PR TITLE
curve: Release 4.1.0

### PR DESCRIPTION
**Pull request by pinkforest**
4.1.0 Adds:
- #555 
- #562 
- #566 

4.1.0 Fixes:
- #558 
- #568 
- #572 

Pending:
- https://github.com/dalek-cryptography/curve25519-dalek/pull/575

### CODE REVIEW

File: curve25519-dalek/README.md [hunk 1]
```diff
@@ -303,8 +308,8 @@ Thanks also to Ashley Hauck, Lucas Salibian, Manish Goregaokar, Jack Grigg,
 Pratyush Mishra, Michael Rosenberg, @pinkforest, and countless others for their
 contributions.
 
-[ed25519-dalek]: https://github.com/dalek-cryptography/ed25519-dalek
-[x25519-dalek]: https://github.com/dalek-cryptography/x25519-dalek
+[ed25519-dalek]: https://github.com/dalek-cryptography/curve25519-dalek/ed25519-dalek
+[x25519-dalek]: https://github.com/dalek-cryptography/curve25519-dalek/x25519-dalek
```
Comment (robjtede) [inline curve25519-dalek/README.md:312]:
```suggestion
[ed25519-dalek]: https://github.com/dalek-cryptography/curve25519-dalek/tree/main/ed25519-dalek
[x25519-dalek]: https://github.com/dalek-cryptography/curve25519-dalek/tree/main/x25519-dalek
```

Comment (robjtede) [reply curve25519-dalek/README.md:312]: also worth changing the manifest `repository`

Comment (pinkforest) [reply curve25519-dalek/README.md]: I thought about that but we usually just redirect to root elsewhere

Comment (robjtede) [reply curve25519-dalek/README.md]: oh yea sorry i totally got confused with that (too used to seeing ed25519)

**Comment by tarcieri** [comment]
Perhaps we could get #566 in as well?

**Comment by pinkforest** [comment]
Yeah I was going to list the PR's that are pending

**Comment by tarcieri** [comment]
Would like to propose this small PR too: https://github.com/dalek-cryptography/curve25519-dalek/pull/575

**Comment by rozbb** [comment]
Great call re guidance for semver-exempted features. I understand it now. Thanks!

**Pull request merged**